Source/player.cpp

```diff
@@ -1120,13 +1120,13 @@ void InitMultiView()
 	ViewY = myPlayer.position.tile.y;
 }
 
-bool SolidLoc(int x, int y)
+bool SolidLoc(Point position)
 {
-	if (x < 0 || y < 0 || x >= MAXDUNX || y >= MAXDUNY) {
+	if (position.x < 0 || position.y < 0 || position.x >= MAXDUNX || position.y >= MAXDUNY) {

```

Comment:
Making a function for checking for level OOB would be a big help to a lot of code as well and probably catch a handful of bugs (we already fixed a lot of issues like that).

Reply:
For sure. Ideally, we want to limit operating on the individual coordinates of these points as much as possible, and these bound checks could easily be done using rectangle-like operations. I'll look into this at some point in the near future when more of the code is properly using points. We also need some Matrix2d abstractions that can be indexed by points.